pkgs/servers/irc/inspircd/default.nix

```diff
@@ -0,0 +1,218 @@
+    ssl_mbedtls     = [ mbedtls ];
+    ssl_openssl     = [ openssl ];
+  };
+
+  # buildInputs necessary for the enabled extraModules
+  extraInputs = lib.concatMap
+    (m: extras."${m}" or (builtins.throw "Unknown extra module ${m}"))
+    extraModules;
+
+  # if true, we can't provide a binary version of this
+  # package without violating the GPL 2
+  gpl2Conflict =
+    let
+      allowed = compatibleModules lib stdenv;
+    in
+      !lib.all (lib.flip lib.elem allowed) extraModules;
+
+  # return list of the license(s) of the given derivation
+  getLicenses = drv:
+    let
+      lics = drv.meta.license or [];
+    in
+      if lib.isAttrs lics || lib.isString lics
+      then [ lics ]
+      else lics;
+
+  # Whether any member of list1 is also member of list2, i. e. set intersection.
+  anyMembers = list1: list2:
+    lib.any (m1: lib.elem m1 list2) list1;
+
+in
+
+stdenv.mkDerivation rec {
+  pname = "inspircd";
+  version = "3.9.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0x3paasf4ynx4ddky2nq613vyirbhfnxzkjq148k7154pz3q426s";
+  };
+
+  patches = [
+    # Remove at next release, makes --prefix and --system work together
+    (fetchpatch {
+      url = "https://github.com/inspircd/inspircd/commit/b378b5087b41f72a1624ebb58990180e0b0140aa.patch";
+      sha256 = "0c0fmhjbkfh2r1cmjrm5d4whcignwsyi6kwkhmcvqy9mv99pj2ir";
+    })
+  ];
+
+  nativeBuildInputs = [
+    perl
+    pkg-config
+  ];
+  buildInputs = extraInputs;
+
+  preConfigure = ''
+    patchShebangs configure make/*.pl
+    # configure is executed twice, once to set the extras
+    # to use and once to do the Makefile setup
+    ./configure --enable-extras \
+      ${lib.escapeShellArg (lib.concatStringsSep " " extraModules)}
+  '';
+
+  configureFlags = [
+    "--disable-auto-extras"
+    "--distribution-label" "nixpkgs${version}"
+    "--system"
+    "--uid" "0"
+    "--gid" "0"
+    "--prefix" (placeholder "out")
+  ];
+
+  postInstall = ''
+    # installs to $out/usr by default unfortunately
+    mv "$out/usr/lib" "$out/lib"
+    mv "$out/usr/sbin" "$out/bin"
+    mv "$out/usr/share" "$out/share"
+    rm -r "$out/usr"
+    rm -r "$out/var" # only empty directories
+    rm -r "$out/etc" # only contains documentation
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = {
+    description = "A modular C++ IRC server";
+    license = [ lib.licenses.gpl2Only ]
+      ++ lib.concatMap getLicenses extraInputs
+      ++ lib.optionals (anyMembers extraModules libcModules) (getLicenses stdenv.cc.libc)
+      # FIXME(sternenseemann): get license of used lib(std)c++ somehow
+      ++ lib.optional (anyMembers extraModules libcxxModules) "Unknown"
+      # Hack: Definitely prevent a hydra from building this package on
+      # a GPL 2 incompatibility even if it is not in a top-level attribute,
+      # but pulled in indirectly somehow.
+      ++ lib.optional gpl2Conflict lib.licenses.unfree;
```

Comment:
What’s the semantics here? All?

Reply:
`hasUnfree` in `checkMeta` checks if any license is unfree and then sets the `meta.unfree` attribute accordingly if that's what you're referring to.

I like doing it this way because it then has all the licenses applying plus the hint that this combination creates a situation where it is unfree in a sense, i. e. we can't redistribute it.